src/ast/ast_validations.c

```diff
@@ -1035,15 +1030,11 @@ static AST_Validation _ValidateUnion_Clauses
 			}
 
 			for(uint j = 0; j < proj_count; j++) {
-				const cypher_astnode_t *proj = cypher_ast_return_get_projection(return_clause, j);
-				const cypher_astnode_t *alias_node = cypher_ast_projection_get_alias(proj);
-				if(alias_node == NULL)  {
-					// The projection was not aliased, so the projection itself must be an identifier.
-					alias_node = cypher_ast_projection_get_expression(proj);
-					ASSERT(cypher_astnode_type(alias_node) == CYPHER_AST_IDENTIFIER);
-				}
-				const char *alias = cypher_ast_identifier_get_name(alias_node);
-				if(strcmp(projections[j], alias) != 0) {
+				const cypher_astnode_t *proj =
+					cypher_ast_return_get_projection(return_clause, j);
+				const char *alias = AST_GetProjectionAlias(proj);
```

Comment:
is it possible for `AST_GetProjectionAlias` to return `NULL` ?

Reply:
No. The unneeded comparison was removed from ``if(alias == NULL || ...``